File: Android/app/src/main/java/catglo/com/deliverydroid/DeliveryDroidBaseActivity.java
package catglo.com.deliverydroid;

import android.app.backup.BackupManager;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import androidx.appcompat.app.AppCompatActivity;
import catglo.com.deliveryDatabase.DataBase;


public class DeliveryDroidBaseActivity extends AppCompatActivity implements Tooled {

    @Override
    public Tools getTools() {
        return tools;
    }

    public interface OnKeyboardVisibilityListener {
        void onVisibilityChanged(boolean visible);
    }
    
    public Tools tools = new Tools();
    public DataBase dataBase = null;
    public SharedPreferences sharedPreferences;

    static boolean unlocked=false;
	public static boolean startNewRun=true;

		
	@Override
	public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
		if (getActionBar()!=null) getActionBar().hide();
        dataBase = tools.create(this);
    	sharedPreferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
    }
	
	 @Override
	  public void onStart() {
	    super.onStart();
	  }

	  @Override
	  public void onStop() {
	    super.onStop();
	  }
	  
	
    @Override
	protected void onResume() {
		super.onResume();
		
		BackupManager b = new BackupManager(this);
		b.dataChanged();
		
		
	}

	@Override
	protected void onPause() {
		//dataBase.close();
		super.onPause();
	}



   





}

File: Android/app/src/main/java/catglo/com/deliverydroid/DeliveryDroidBaseActivity.kt
package catglo.com.deliverydroid

import android.app.backup.BackupManager
import android.content.*
import android.os.Build
import android.os.Bundle
import android.os.IBinder
import android.preference.PreferenceManager
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import catglo.com.MapFileSharingService
import catglo.com.deliveryDatabase.DataBase
import java.lang.IllegalStateException


open class DeliveryDroidBaseActivity : AppCompatActivity(), Tooled , ServiceConnection {

    public var mapSharingServce : MapFileSharingService? = null

    override fun onServiceDisconnected(name: ComponentName?) {
        mapSharingServce = null
    }

    override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
        (service as MapFileSharingService.LocalBinder)?.let {
            mapSharingServce = it.getService()
        }
    }

    var tools = Utils()
    lateinit var dataBase: DataBase
    lateinit var sharedPreferences: SharedPreferences

    override fun getUtils(): Utils {
        return tools
    }

    private var isBound = false
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(applicationContext)
        dataBase = tools.create(this)
    }

    override fun onResume() {
        super.onResume()
        val b = BackupManager(this)
        b.dataChanged()
        if (Settings(this).mapDownloadOption == MapDownloadOption.torrent) {
            try {
                val serviceIntent = Intent(this, MapFileSharingService::class.java)
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                    startForegroundService(serviceIntent)
                } else {
                    startService(serviceIntent)
                }
                bindService(serviceIntent, this, Context.BIND_AUTO_CREATE)
                isBound = true
            } catch (e:IllegalStateException){
                Log.e("Torrent","Failed to start download service "+e.localizedMessage)
            }
        }
    }

    override fun onPause() {
        super.onPause()
        if (isBound)
        {
            isBound = false
            unbindService(this)
            mapSharingServce = null
        }
    }

    companion object {
        var startNewRun = true
    }
}
